File: server/data/results/translations/anchored/go/fibonacci_memo.go
package main

import "fmt"

func fib(n int, memo map[int]int) int {
	if n <= 1 {
		return n
	}
	if val, ok := memo[n]; ok {
		return val
	}
	value := fib(n-1, memo) + fib(n-2, memo)
	memo[n] = value
	return value
}

func main() {
	memo := make(map[int]int)
	for i := 0; i < 15; i++ {
		fmt.Println(fib(i, memo))
	}
}

File: server/data/results/translations/anchored/java/fibonacci_memo.java
import java.util.HashMap;
import java.util.Map;

public class Main {

    public static int fib(int n, Map<Integer, Integer> memo) {
        if (n <= 1) {
            return n;
        }
        if (memo.containsKey(n)) {
            return memo.get(n);
        }
        int value = fib(n - 1, memo) + fib(n - 2, memo);
        memo.put(n, value);
        return value;
    }

    public static void main(String[] args) {
        Map<Integer, Integer> memo = new HashMap<>();
        for (int i = 0; i < 15; i++) {
            System.out.println(fib(i, memo));
        }
    }
}